frame/offences/src/tests.rs

```diff
@@ -71,14 +75,34 @@ fn should_not_report_the_same_authority_twice_in_the_same_slot() {
 	});
 }
 
+#[test]
+fn should_not_report_an_obsolete_offence() {
+	new_test_ext().execute_with(|| {
+		// given
+		let time_slot = 42;
+		let session_index = 3;
+		assert_eq!(offence_reports(KIND, time_slot), vec![]);
+
+		let offence =
+			Offence { validator_set_count: 5, session_index, time_slot, offenders: vec![5] };
+		assert_eq!(Offences::report_offence(vec![], offence), Err(OffenceError::ObsoleteReport));
+
+		with_on_offence_fractions(|f| {
```

Comment:
Not sure why we use this convoluted helper function here. Isn't it equal to `assert!(OnOffencePerbill::get().is_empty())`?

Reply:
using the helper function to just assert equality is a bit weird to me.

Reply:
or else at least a `const SESSION_INDEX:u32 = 10;`